Detectors/CTP/workflow/src/RawToDigitConverterSpec.cxx

```diff
@@ -0,0 +1,195 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <string>
+#include "FairLogger.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "Framework/InputRecordWalker.h"
+#include "Framework/WorkflowSpec.h"
+#include "DetectorsRaw/RDHUtils.h"
+#include "DPLUtils/DPLRawParser.h"
+#include "CTPWorkflow/RawToDigitConverterSpec.h"
+
+using namespace o2::ctp::reco_workflow;
+
+void RawToDigitConverterSpec::init(framework::InitContext& ctx)
+{
+}
+
+void RawToDigitConverterSpec::run(framework::ProcessingContext& ctx)
+{
+  mOutputDigits.clear();
+  std::map<o2::InteractionRecord, CTPDigit> digits;
+  const gbtword80_t bcidmask = 0xfff;
+  gbtword80_t pldmask;
+  using InputSpec = o2::framework::InputSpec;
+  using ConcreteDataTypeMatcher = o2::framework::ConcreteDataTypeMatcher;
+  using Lifetime = o2::framework::Lifetime;
+  //mOutputHWErrors.clear();
+  std::vector<InputSpec> filter{InputSpec{"filter", ConcreteDataTypeMatcher{"CTP", "RAWDATA"}, Lifetime::Timeframe}};
+  o2::framework::DPLRawParser parser(ctx.inputs(), filter);
+  //setUpDummyLink
+  auto& inputs = ctx.inputs();
+  // if we see requested data type input with 0xDEADBEEF subspec and 0 payload this means that the "delayed message"
+  // mechanism created it in absence of real data from upstream. Processor should send empty output to not block the workflow
+  {
+    std::vector<InputSpec> dummy{InputSpec{"dummy", o2::framework::ConcreteDataMatcher{"CTP", "RAWDATA", 0xDEADBEEF}}};
+    for (const auto& ref : o2::framework::InputRecordWalker(inputs, dummy)) {
+      const auto dh = o2::framework::DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+      if (dh->payloadSize == 0) {
+        LOGP(WARNING, "Found input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : assuming no payload for all links in this TF",
+             dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+        return;
```

Comment:
See this comment: https://github.com/AliceO2Group/AliceO2/pull/6457#discussion_r653807498
You should push an empty output in this case.

Reply:
added output, pls, check